src/test/ExtractionClient.test.ts

```diff
@@ -0,0 +1,144 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import * as chaiAsPromised from "chai-as-promised";
+import { expect, use } from "chai";
+import * as sinon from "sinon";
+import { ExtractionClient, ExtractionLog, ExtractorState } from "../reporting";
+use(chaiAsPromised);
+
+describe("Extraction Client", () => {
+  const extractionClient: ExtractionClient = new ExtractionClient();
+  const extractionClientNewBase: ExtractionClient = new ExtractionClient("BASE");
+  let fetchStub: sinon.SinonStub;
+  let requestStub: sinon.SinonStub;
+
+  beforeEach(() => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    fetchStub = sinon.stub(ExtractionClient.prototype, "fetchJSON" as any);
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    requestStub = sinon.stub(ExtractionClient.prototype, "createRequest" as any);
+    requestStub.returns("pass");
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  it("run extraction", async function () {
+    const returns = {
+      run: {
+        id: 1,
+      },
+    };
+    fetchStub.resolves(returns);
+    let extraction = await extractionClient.runExtraction("-", "-");
+    expect(extraction.id).to.be.eql(1);
+    expect(fetchStub.calledWith(
+      "https://api.bentley.com/insights/reporting/datasources/imodels/-/extraction/run",
```

Comment:
@MindaugasButkus Need your opinion. Would it be too much to replace this with `${REPORTING_BASE_PATH}/datasources/imodels/-/extraction/run`? I think it will look cleaner and future proof a bit.

Reply:
I think tests should have the full hardcoded path for assertions.